### PR TITLE
Make Helm Chart support Harbor 1.5.0

### PR DESCRIPTION
**Pull request by reasonerjt**
Update the helm chart of Harbor to use 1.5 image.  Specifically
`1.5.0-chart-patch` to include the fix to #4496 (for integration reason
this fix will not be merge to release branch).
Verified against kubenetes v1.10.0

### CODE REVIEW

File: contrib/helm/harbor/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: harbor
-version: 0.1.2
-appVersion: 1.4.0
+version: 0.1.3
```
Comment (jessehu) [inline contrib/helm/harbor/Chart.yaml]: How about bump the version to 0.2.0 ?

File: contrib/helm/harbor/templates/adminserver/adminserver-cm.yaml [hunk 2]
```diff
@@ -54,6 +54,7 @@ data:
   CLAIR_DB_PORT: "5432"
   CLAIR_DB: "{{ .Values.clair.postgresDatabase }}"
   CLAIR_DB_USERNAME: "{{ .Values.clair.postgresUser }}"
+  CLAIR_URL: "http://clair:6060"
```
Comment (jessehu) [inline contrib/helm/harbor/templates/adminserver/adminserver-cm.yaml]: It should be `http://{{ template "harbor.fullname" . }}-clair:6060`

File: contrib/helm/harbor/templates/clair/clair-dpl.yaml [hunk 3]
```diff
@@ -20,7 +20,7 @@ spec:
     spec:
       containers:
       - name: clair
-        image: {{ .Values.clair.image.repository }}:{{ .Values.clair.image.tag }}
+        image: {{ .Values.clair.image.repository }}:{{ .Values.clair.image.tag }}
```
Comment (jessehu) [inline contrib/helm/harbor/templates/clair/clair-dpl.yaml]: pls remove the trailing spaces

File: contrib/helm/harbor/templates/clair/clair-dpl.yaml [hunk 4]
```diff
@@ -20,7 +20,7 @@ spec:
     spec:
       containers:
       - name: clair
-        image: {{ .Values.clair.image.repository }}:{{ .Values.clair.image.tag }}
+        image: {{ .Values.clair.image.repository }}:{{ .Values.clair.image.tag }} 
         imagePullPolicy: {{ .Values.clair.image.pullPolicy }}
         args: ["-insecure-tls", "-config", "/config/config.yaml"]
```
Comment (jessehu) [inline contrib/helm/harbor/templates/clair/clair-dpl.yaml]: should it be /etc/clair/config.yaml ?

File: contrib/helm/harbor/templates/ui/ui-dpl.yaml [hunk 5]
```diff
@@ -38,8 +38,6 @@ spec:
             value: "1"
           - name: ADMINSERVER_URL
             value: "http://{{ template "harbor.fullname" . }}-adminserver"
-          - name: CLAIR_DB_HOST
```
Comment (jessehu) [inline contrib/helm/harbor/templates/ui/ui-dpl.yaml]: is this not needed in 1.5.0?

File: contrib/helm/harbor/values.yaml [hunk 6]
```diff
@@ -26,10 +26,10 @@
 #       cpu: 100m
 
 persistence:
-  enabled: false
+  enabled: true
```
Comment (jessehu) [inline contrib/helm/harbor/values.yaml]: If this enabled by default, users need to config all the storageClass sections. How about keep it as false by default?

File: contrib/helm/harbor/values.yaml [hunk 7]
```diff
@@ -301,17 +302,30 @@ postgresql:
...
+## see https://github.com/kubernetes/charts/tree/master/stable/redis
+## for further configurables.
+redis:
+# Update needed in the cm that defines redis_url if usePassword is set to true.
+  usePassword: false
+  cluster:
+    enabled: false
+  master:
+    persistence:
+# temp workaround the permission issue.
```
Comment (jessehu) [inline contrib/helm/harbor/values.yaml]: suggestion add a TODO here, and explain the details for permission issue (e.g. a bug url)

File: contrib/helm/harbor/templates/ui/ui-dpl.yaml (hunk 5 above)
Comment (reasonerjt) [reply contrib/helm/harbor/templates/ui/ui-dpl.yaml]: Will call admin server to get the value.

File: contrib/helm/harbor/templates/adminserver/adminserver-cm.yaml (hunk 2 above)
Comment (reasonerjt) [reply contrib/helm/harbor/templates/adminserver/adminserver-cm.yaml]: It's not:
https://github.com/vmware/harbor/blob/master/contrib/helm/harbor/templates/clair/clair-svc.yaml#L8

We'll fix this in further refactory to make things more consistent.

File: contrib/helm/harbor/values.yaml (hunk 6 above)
Comment (reasonerjt) [reply contrib/helm/harbor/values.yaml]: I had to set it to true to pass the test when using local storage:
https://github.com/vmware/harbor/blob/master/contrib/helm/harbor/templates/registry/registry-ss.yaml#L55

File: contrib/helm/harbor/templates/clair/clair-dpl.yaml (hunk 4 above)
Comment (reasonerjt) [reply contrib/helm/harbor/templates/clair/clair-dpl.yaml]: yes

File: contrib/helm/harbor/values.yaml (hunk 6 above)
Comment (jessehu) [reply contrib/helm/harbor/values.yaml]: Do you mean if set to false the chart deployment will fail ?

File: contrib/helm/harbor/Chart.yaml (hunk 1 above)
Comment (jessehu) [reply contrib/helm/harbor/Chart.yaml]: How about bump the version to 0.2.0 ?  It should be at least a minor version bump.

File: contrib/helm/harbor/values.yaml (hunk 6 above)
Comment (reasonerjt) [reply contrib/helm/harbor/values.yaml]: Correct.

File: contrib/helm/harbor/Chart.yaml (hunk 1 above)
Comment (reasonerjt) [reply contrib/helm/harbor/Chart.yaml]: ok

**Comment by reasonerjt** [comment]
Updated the version of chart and the commit message according to review comment.

**Pull request merged**